diff -Nur sh-utils-2.0.ORIG/src/Makefile.am sh-utils-2.0/src/Makefile.am
--- sh-utils-2.0.ORIG/src/Makefile.am	Wed Aug 16 21:49:21 2000
+++ sh-utils-2.0/src/Makefile.am	Wed Aug 16 22:28:15 2000
@@ -8,7 +8,7 @@
 bin_SCRIPTS = groups @OPTIONAL_BIN_ZCRIPTS@
 
 bin_PROGRAMS = basename date dirname echo env expr factor false \
-  hostname id logname pathchk pinky printenv printf pwd seq \
+  hostname id logname pathchk pinky printenv printf pwd seq shuffle \
   sleep tee test true tty users who whoami yes @OPTIONAL_BIN_PROGS@
 
 datadir = $(prefix)/@DATADIRNAME@
diff -Nur sh-utils-2.0.ORIG/src/shuffle.c sh-utils-2.0/src/shuffle.c
--- sh-utils-2.0.ORIG/src/shuffle.c	Wed Dec 31 19:00:00 1969
+++ sh-utils-2.0/src/shuffle.c	Wed Aug 16 22:37:11 2000
@@ -0,0 +1,96 @@
+/* shuffle - shuffles argument list and prints to stdout
+   Copyright (C) 1991-1997, 1999 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+/* Ruda Moura <ruda@helllabs.org> */
+
+#include <config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#include <getopt.h>
+
+#include "system.h"
+#include "long-options.h"
+
+/* The official name of this program (e.g., no `g' prefix).  */
+#define PROGRAM_NAME "shuffle"
+
+#define AUTHORS "Ruda Moura"
+
+/* The name this program was run with. */
+char *program_name;
+
+void
+usage (int status)
+{
+  if (status != 0)
+    fprintf (stderr, _("Try `%s --help' for more information.\n"),
+	     program_name);
+  else
+    {
+      printf (_("Usage: %s [OPTION]... [ARGUMENT]...\n"), program_name);
+      printf (_("\
+Shuffles ARGUMENT list and prints to standard output.\n\
+\n\
+  --help      display this help and exit\n\
+  --version   output version information and exit\n"));
+      puts (_("\nReport bugs to <bug-sh-utils@gnu.org>."));
+    }
+  exit (status);
+}
+
+int
+main (int argc, char **argv)
+{
+  int i, j, k;
+  int n = argc - 1;
+  char *tmp;
+  struct timeval tv;
+  struct timezone tz;
+
+  program_name = argv[0];
+  setlocale (LC_ALL, "");
+  bindtextdomain (PACKAGE, LOCALEDIR);
+  textdomain (PACKAGE);
+
+  /* Don't recognize --help or --version if POSIXLY_CORRECT is set.  */
+  if (getenv ("POSIXLY_CORRECT") == NULL)
+    parse_long_options (argc, argv, PROGRAM_NAME, GNU_PACKAGE, VERSION,
+			AUTHORS, usage);
+
+  if (argc == 1)
+    return 0;
+
+  /* seed */
+  gettimeofday (&tv, &tz);
+  srand (tv.tv_usec);
+
+  /* shuffle */
+  for (i=0; i < n; i++) {
+    j = (rand () % n) + 1;
+    k = (rand () % n) + 1;
+    tmp = argv[j];
+    argv[j] = argv[k];
+    argv[k] = tmp;
+  }    
+
+  /* print */
+  for (i=1; i < n; i++)
+    printf ("%s ", argv[i]);
+  printf ("%s\n", argv[i]);
+}
diff -Nur sh-utils-2.0.ORIG/po/POTFILES.in sh-utils-2.0/po/POTFILES.in
--- sh-utils-2.0.ORIG/po/POTFILES.in	Wed Nov  8 14:18:44 2000
+++ sh-utils-2.0/po/POTFILES.in	Wed Nov  8 17:06:04 2000
@@ -34,6 +34,7 @@
 src/printf.c
 src/pwd.c
 src/seq.c
+src/shuffle.c
 src/sleep.c
 src/stty.c
 src/su.c
